app/trace/trace_viewer.tsx

```diff
@@ -84,6 +89,10 @@ export default class TraceViewer extends React.Component<TraceViewProps, {}> {
         passive: false,
       });
     }
+    if (this.filterInputRef.current) {
+      this.filterInputRef.current.addEventListener("keyup", () => this.updateFilter());
```

Comment:
would it work to write

`<FilterInput onChange={(e) => this.updateFilter(e.target.value)} ... />`

instead of using a ref + addEventListener? (I don't think it would trigger a full re-render of this TraceViewer component on each keystroke, but could be mistaken)

Reply:
Nice, done. I initially went this way - but switched because I wasn't using state. But now that I'm driving everything off the URL param it's pretty clean.